crates/bin/pcli/src/command/tx/auction/dutch/mod.rs

```diff
@@ -0,0 +1,448 @@
+use penumbra_keys::keys::AddressIndex;
+use penumbra_num::Amount;
+use penumbra_proto::{view::v1::GasPricesRequest, DomainType, Name};
+use penumbra_view::SpendableNoteRecord;
+use penumbra_view::ViewClient;
+use penumbra_wallet::plan::Planner;
+use rand::RngCore;
+use rand_core::OsRng;
+use serde_json;
+
+mod debug;
+pub mod gda;
+
+/// Commands related to Dutch auctions
+#[derive(Debug, Subcommand)]
+pub enum DutchCmd {
+    /// Schedule a gradual dutch auction, a prototype for penumbra developers.
+    #[clap(display_order = 1000, name = "gradual")]
+    DutchAuctionGradualSchedule {
+        /// Source account initiating the auction.
+        #[clap(long, display_order = 100, default_value = "0")]
+        source: u32,
+        /// The value the seller wishes to auction.
+        #[clap(long, display_order = 200)]
+        input: String,
+        /// The maximum output the seller can receive.
+        ///
+        /// This implicitly defines the starting price for the auction.
+        #[clap(long, display_order = 400)]
+        max_output: String,
+        /// The minimum output the seller is willing to receive.
+        ///
+        /// This implicitly defines the ending price for the auction.
+        #[clap(long, display_order = 500)]
+        min_output: String,
+        /// The duration for the auction
+        #[clap(arg_enum, long, display_order = 600, name = "duration")]
+        recipe: gda::GdaRecipe,
+        /// Skip asking for confirmation, pay any fees, and execute the transaction.
+        #[clap(long, display_order = 700)]
+        yes: bool,
+        /// The selected fee tier to multiply the fee amount by.
+        #[clap(short, long, value_enum, default_value_t, display_order = 1000)]
+        fee_tier: FeeTier,
+        #[clap(long, hide = true)]
+        // Use to produce a debug file for numerical analysis.
+        debug: bool,
+    },
+    /// Schedule a Dutch auction, a tool to help accomplish price discovery.
+    #[clap(display_order = 100, name = "schedule")]
+    DutchAuctionSchedule {
+        /// Source account initiating the auction.
+        #[clap(long, display_order = 100, default_value = "0")]
+        source: u32,
+        /// The value the seller wishes to auction.
+        #[clap(long, display_order = 200)]
+        input: String,
+        /// The maximum output the seller can receive.
+        ///
+        /// This implicitly defines the starting price for the auction.
+        #[clap(long, display_order = 400)]
+        max_output: String,
+        /// The minimum output the seller is willing to receive.
+        ///
+        /// This implicitly defines the ending price for the auction.
+        #[clap(long, display_order = 500)]
+        min_output: String,
+        /// The block height at which the auction begins.
+        ///
+        /// This allows the seller to schedule an auction at a future time.
+        #[clap(long, display_order = 600)]
+        start_height: u64,
```

Comment:
https://docs.rs/clap/latest/clap/_derive/index.html#command-attributes

clap provides a `flatten` attribute that seems like it would be really useful for the standard / gradual auction commands.

```rust
/// Doc comment
#[derive(Parser)]
#[command(CMD ATTRIBUTE)]
#[group(GROUP ATTRIBUTE)]
struct Cli {
    #[arg(short)]
    field: SomeType,
    #[command(flatten)]
    dutch_args: DutchArgs,
}

/// Doc comment
#[derive(Args)]
#[command(PARENT CMD ATTRIBUTE)]
#[group(GROUP ATTRIBUTE)]
struct DutchArgs {
    /// Doc comment
    #[command(ARG ATTRIBUTE)]
    common_field: (),
}
```

paraphrasing roughly from the clap docs, we could do something like this to refrain from repeating the same arguments twice. this feels worthwhile because it will mean that the distinctions between the two auction variants will be more visible.

Reply:
punting, "gradual" auction support is an r&d tool.